### PR TITLE
Collect must-gather OCS logs when a worker node is down

### PR DESCRIPTION
**Pull request by OdedViner**
Signed-off-by: Oded Viner <oviner@redhat.com>
Polarion ID:
https://polarion.engineering.redhat.com/polarion/#/project/OpenShiftContainerStorage/workitem?id=OCS-2328
BZ:
https://bugzilla.redhat.com/show_bug.cgi?id=1770199

### CODE REVIEW

File: tests/manage/z_cluster/test_must_gather.py [hunk 1]
```diff
@@ -70,10 +77,39 @@ def finalizer():
...
+    @pytest.mark.polarion_id("OCS-1583")
     def test_must_gather(self):
         """
         Tests functionality of: oc adm must-gather
 
+        """
+        self.collect_must_gather()
+
+    @pytest.mark.polarion_id("OCS-2328")
+    def test_must_gather_worker_node_down(self, nodes):
```
Comment (ebenahar) [inline tests/manage/z_cluster/test_must_gather.py:89]: this class has `tier1` decorator. This test should not be part of tier1 but rather tier4 IMO.

File: tests/manage/z_cluster/test_must_gather.py [hunk 2]
```diff
@@ -70,10 +77,39 @@ def finalizer():
...
+        # Stop one worker node
+        nodes.stop_nodes(get_node_objs([worker_node]))
+
+        # Collect must gather and check content
+        self.collect_must_gather()
+
+        # Start worker node
+        nodes.start_nodes(get_node_objs([worker_node]))
+
+    def collect_must_gather(self):
```
Comment (AviadP) [inline tests/manage/z_cluster/test_must_gather.py:109]: This name does not describe the function - i would got for something like `run_must_gether_test`, as it this main testing function

Comment (OdedViner) [reply tests/manage/z_cluster/test_must_gather.py]: `validate_must_gather_content` is ok?

File: tests/manage/z_cluster/test_must_gather.py (hunk 1 above)
Comment (OdedViner) [reply tests/manage/z_cluster/test_must_gather.py:90]: `test_must_gather_worker_node_down` -> tier4a
`test_must_gather` -> tier1

**Review by AviadP — APPROVED** [review]

File: tests/manage/z_cluster/test_must_gather.py (hunk 1 above)
Comment (ebenahar) [reply tests/manage/z_cluster/test_must_gather.py:90]: +1

**Review by ebenahar — DISMISSED** [review]

**Review by fbalak — DISMISSED** [review]

File: tests/manage/z_cluster/test_must_gather.py [hunk 3]
```diff
@@ -70,10 +76,40 @@ def finalizer():
...
+        self.reboot_node = True
+        logger.info('Get all worker nodes and choose random from worker nodes list')
+        worker_nodes = get_worker_nodes()
+        worker_node = worker_nodes[randint(0, len(worker_nodes) - 1)]
+
+        # Stop one worker node
+        nodes.stop_nodes(get_node_objs([worker_node]))
+
+        # Collect must gather and check content
+        self.collect_must_gather()
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/test_must_gather.py:104]: must-gather pod will fail to come if the pod tries to come on the down node.
We should pass --node-name in this case where one or more worker nodes are not in Ready state. 

 oc adm must-gather --image=registry.redhat.io/ocs4/ocs-must-gather-rhel8 --dest-dir=<directory-name> --node-name=<node-name>

**Pull request closed without merging**